<?xml version="1.0"?>
<launch>
    <!-- Parameter file -->
    <arg name="params_file" default="$(find simulation_env_ros1)/config/params.yaml"/>
    <arg name="rviz_config" default="$(find simulation_env_ros1)/rviz/simulation.rviz"/>

    <!-- simulation_node -->
    <node name="simulation_node" pkg="simulation_env_ros1" type="simulation_node" output="screen">
        <!-- Load parameters to node's private namespace -->
        <rosparam file="$(arg params_file)" command="load" subst_value="true"/>
    </node>

    <!-- simulated_lidar -->
    <node name="simulated_lidar" pkg="simulation_env_ros1" type="simulated_lidar" output="screen">
        <!-- LiDAR parameters -->
        <param name="horizontal_resolution" value="0.05"/>
        <param name="num_laser_lines" value="64"/>
        <param name="max_range" value="5.0"/>
        <param name="min_range" value="0.1"/>
        <param name="noise_stddev" value="0.02"/>
        <param name="robot_frame" value="base_link"/>
        <param name="world_frame" value="map"/>
        <param name="publish_rate" value="10.0"/>
        <param name="tf_lookup_timeout" value="0.1"/>
        <!-- Boundary check parameters (fix clipping issues) -->
        <param name="enable_boundary_check" value="true"/>
    </node>

    <!-- RViz visualization -->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rviz_config)" output="screen"/>
</launch>

